Hm 4

SunnatBakidjanov opened the pull request.

SunnatBakidjanov committed:
'Разделены компоненты'

--UI коунтера в Dish.jsx выведена в отдельный компонент
--Создан компонент Counter.jsx
--Создан компонент DishCounter.jsx
--Создана папка hooks
--Хуки useCount.js и useTabs.js добавлены в папку hooks

SunnatBakidjanov committed:
'Переделана логика в useCount'

SunnatBakidjanov committed:
'Добавлен компонент с логикой ReviewForm.jsx'

SunnatBakidjanov committed:
'Изменена логика валидации ресторана'

SunnatBakidjanov committed:
'Добавлен логика форм'

--Добавлен хук useForm
--Доработана форма

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -1,29 +1,34 @@
+                    <li key={restaurants.id}>
+                        <RestaurantItem {...validatedRestaurant} />
+
+                        <ReviewForm />
```
форму отзыва необходимо унести в RestaurantItem

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -1,29 +1,34 @@
-                    <li key={findRestaurant.id}>
-                        <RestaurantItem {...validateRestaurant(findRestaurant)} />
+            {validatedRestaurant ? (
+                <ul className='restaurants-list'>
+                    <li key={restaurants.id}>
```
тут не нужен список, так как мы отображаем 1 элемент

zubkov7 commented on src/components/restaurants/review-form/ReviewForm.jsx:
```diff
@@ -0,0 +1,34 @@
+        <>
+            <h3>Форма</h3>
+
+            <form className='form' action='#'>
```
чтобы не срабатывала перезагрузка страницы, можно добавить проп `onSubmit={e => e.preventDefault()}`

SunnatBakidjanov committed:
'Внесены правки'

--ReviewForm перенесена в RestaurantItem
--Удален список и добавлен div для отображения 1 элемента
--Добавлен проп onSubmit в форму ReviewForm

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -19,13 +18,9 @@ export const Restaurants = () => {
-                        <ReviewForm />
-                    </li>
-                </ul>
+                <>
```
фрагмент не нужен

SunnatBakidjanov committed:
Убран фрагмент из Restaurants
